### PR TITLE
Add documentation-builder to 1.25 docs

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
- Added a build script
- Created the navigation yaml
- Moved the media to the src directory
- Updated links to the images in the docs

## QA
- Run `./build-docs.sh`
- Then `cd build`
- Run a server, for example: `python -m SimpleHTTPServer 8000`
- Go to http://localhost:8000/en/config-aws.html
- Look around and check it all works

### CODE REVIEW

**Comment by evilnick** [comment]
suggestion - don't move all the media files! just create a symlink

      cd src
      ln -s ../htmldocs/media media

this removes the need to break current jujucharms build, cuts the diff considerably

File: src/en/authors-charm-components.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ metadata, configuration data, and hooks with some extra support files.
 
 A charm requires only a single file in order to be considered valid by juju:
 
- - `metadata.yaml` [describes the charm](./authors-charm-metadata.html) and the
+ - `metadata.yaml` [describes the charm](../authors-charm-metadata.html) and the
```
Comment (evilnick) [inline src/en/authors-charm-components.md]: I don't think this is the path we want, it should be just
      [describes the charm](authors-charm-metadata.md)
or
     [describes the charm](./authors-charm-metadata.md)

File: src/en/authors-charm-metadata.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ Here's a valid metadata file:
     description: |
       MongoDB is a high-performance, open source, schema-free document- oriented
       data store that's easy to deploy, manage and use. It's network accessible,
-      written in C++ and offers the following features:  
+      written in C++ and offers the following features:
```
Comment (evilnick) [inline src/en/authors-charm-metadata.md]: I think covered in a different review, we don't want to strip all the trailing spaces - some of them have meaning

File: tools/mdbuild.py [hunk 3]
```diff
@@ -143,7 +143,7 @@ def post_process(self):
             ('%%CONTENT%%', self.parsed),
             ('%%DOCNAV%%', doc_nav),
             ('src="media/', 'src="../media/'),
-            ('src="./media/', 'src="../media/'),
+            ('src="../media/', 'src="../media/'),
```
Comment (evilnick) [inline tools/mdbuild.py]: don't change this

File: src/en/authors-charm-components.md (hunk 1 above)
Comment (anthonydillon) [reply src/en/authors-charm-components.md]: Good point, I'll revert the html links and leave the image links.

File: src/en/authors-charm-metadata.md (hunk 2 above)
Comment (anthonydillon) [reply src/en/authors-charm-metadata.md]: What are they used for? I have checked both versions and they look the same.

File: tools/mdbuild.py (hunk 3 above)
Comment (anthonydillon) [reply tools/mdbuild.py]: :+1:

**Comment by anthonydillon** [comment]
I copied the images for now as we want to remove the htmldocs directory once we go live with the new system. I didn't want use to forgot but it is more maintenance so will symlink for now. Just need to remember when removing the htmldocs directory to move the media directory.

File: src/en/authors-charm-metadata.md (hunk 2 above)
Comment (evilnick) [reply src/en/authors-charm-metadata.md]: two trailing spaces insert a `<br>`. It is often a neater way to do it that to include html in the source or to mess the format up with hard carraige returns (which can cause other problems in markdown). They are pretty much essential in lists, as its the only way to force a linebreak without ending the list. So, although many of them are just cruft, some are there on purpose. It's just easier to disable any trailing space deletion, or people get [upset about it](https://github.com/Microsoft/vscode/issues/1679)

**Comment by evilnick** [comment]
don't worry, we won't forget about them :smiley:

**Comment by anthonydillon** [comment]
I'll re-do the content so it retains whitepspacing. Will try and get to that tonight.

**Comment by anthonydillon** [comment]
@evilnick Ok think I have reverted all the whitespace changes.

File: tools/mdbuild.py [hunk 4]
```diff
@@ -143,7 +143,7 @@ def post_process(self):
             ('%%CONTENT%%', self.parsed),
             ('%%DOCNAV%%', doc_nav),
             ('src="media/', 'src="../media/'),
-            ('src="./media/', 'src="../media/'),
+            ('src="./media/', 'src="./media/'),
```
Comment (evilnick) [inline tools/mdbuild.py]: :laughing: now you've changed the other side of this! Leave poor mdbuild.py alone!

**Comment by evilnick** [comment]
This looks promising! i will give it a proper run through in the morning

**Comment by anthonydillon** [comment]
Woops, my bad. reverted `mdbuild.py`. Cheers @evilnick

**Pull request merged**